src/prefect/engine.py

```diff
@@ -737,8 +735,10 @@ def enter_task_run_engine(
 
     # Sync task run in async flow run
     else:
-        # Call out to the sync portal since we are not in a worker thread
-        return flow_run_context.sync_portal.call(begin_run)
+        # Submit to background tasks since we are not in a worker thread
+        flow_run_context.background_tasks.start_soon(begin_run)
+        # get futures from the context
+        return flow_run_context.task_run_futures
```

Comment:
These should be appended and available from https://github.com/PrefectHQ/prefect/pull/7041/files#diff-ae6f1b6d423c9f28188aedf4ab0f0afbce428d61b84feed78ca9c716ce3e26caL919

However, when I execute a flow, it's empty when returned. I think it has something to do with begin_run running in the background and results not yet available.

Reply:
Yeah, these are not populated because your function has been sent to the background

> I haven't found a way to await the results in a sync function.

That's the whole point of the blocking portal :D

Reply:
Right, but if the blocking portal and background tasks are in a separate loop, it leads to:
```
Right, but I based it off
```
RuntimeError: Task <Task pending name='' coro=<create_task_run_then_submit() running at /Users/mz/dev/prefect/src/prefect/engine.py:940> cb=[TaskGroup._spawn.<locals>.task_done() at /opt/homebrew/Caskroom/miniconda/base/envs/orion-dev-38/lib/python3.8/site-packages/anyio/_backends/_asyncio.py:726]> got Future <Future pending> attached to a different loop
```
```

Reply:
That looks like we're incorrectly attempting to wait for the result of a future from a different loop than it was created in. Futures need to consistently belong to a single event loop. We have some definite issues with this and it's proven a bit complicated to untangle.

Reply:
Thanks for the insight.

Put this in `enter_task_run_engine` and `begin_task_map` and reverting back to sync_portal:
```
    import asyncio
    loop = asyncio.get_running_loop()
    loop.set_debug(True)
    print(id(loop))
```

Results in two separate loop IDs
```
4445688688
4765149456
```

So I wonder if there's a way to pass the existing event loop `sync_portal`

Reply:
The whole point of the sync portal is to open a new event loop in a new thread. If we use the existing loop, when we send something to it it'd deadlock as the main loop in the main thread waits for the result of the call in the portal's thread which needs to execute code in the main loop.

Reply:
(I know this hurts the brain — we can find some time for a sync chat about this soon)